lib/payment_calculator/ecf/uplift_calculation.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require "payment_calculator/ecf/contract/uplift_payment_calculations"
+
+module PaymentCalculator
+  module Ecf
+    class UpliftCalculation
+      include Ecf::Contract::UpliftPaymentCalculations
+
+      # @param [Symbol] event_type
+      # @param [Integer] uplift_participants
+      # This is end number of participants who will be used to make the payment calculation.
+      # All invalid users will have already been filtered out before this number is generated and passed here.
+      def call(uplift_participants: 0, event_type: :started)
```

Comment:
`Ecf::Contract::UpliftPaymentCalculations` has `HasDIParameters` included in it, and I vaguely remember that that module has its own idea about what a `call` of a class should do. How does it interact with this `call`?

And, well, should we maybe not mix the two approaches?

Reply:
The class level call was designed to work in this way. The `uplift_participants` and `event_type` will be passed to this instance call since they are named parameters in the class level call.